### PR TITLE
898110: Remove WMP dependencies; Queue declare fix

### PR DESCRIPTION
**Pull request by raghavendran-gopalakrishnan**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=898110

### CODE REVIEW

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/StagingQueueRerouter.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * Create a staging queue name from the 
+     * @param targetQueue The original queue name
+     * @param tenant The tenant id
+     * @return A new queue name that combines the original queue name with the tenant id
+     */
+    public static String route(final String targetQueue, final String tenant) {
+        return targetQueue + LOAD_BALANCED_INDICATOR + "/" + tenant;
+    }
+}
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/StagingQueueRerouter.java]: New line

**Review by andyreidz — CHANGES_REQUESTED** [review]

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java [hunk 2]
```diff
@@ -102,8 +101,11 @@ public static QueueServices create(final String targetQueue, final String partit
         }
 
         //  Declare worker queue.
-        LOG.debug("Passively declaring worker queue {}...", stagingQueueOrTargetQueue);
-        publishChannel.queueDeclarePassive(stagingQueueOrTargetQueue);
+        LOG.debug("Declaring worker queue {}...", stagingQueueOrTargetQueue);
+        final Map<String, Object> queueArguments = new HashMap<>();
+        queueArguments.put("x-queue-type", "quorum");
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]: These need to come from configuration, much like how workers have their queues configured.
You have to handle classic queues with a max-priority and quorum without priority.

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/StagingQueueRerouter.java (hunk 1 above)
Comment (raghavgopal) [reply job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/StagingQueueRerouter.java]: Addressed

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java (hunk 2 above)
Comment (raghavgopal) [reply job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]: Addressed

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java [hunk 3]
```diff
@@ -59,6 +60,12 @@ public final class QueueServicesFactory
                     "CAF_WMP_TARGET_QUEUE_NAMES_PATTERN must be set if CAF_WMP_ENABLED is true"))
             : null;
 
+    public static final String RABBIT_PROP_KEY_MAX_PRIORITY = "x-max-priority";
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]: These can be private

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java [hunk 4]
```diff
@@ -102,8 +107,16 @@ public static QueueServices create(final String targetQueue, final String partit
         }
 
         //  Declare worker queue.
-        LOG.debug("Passively declaring worker queue {}...", stagingQueueOrTargetQueue);
-        publishChannel.queueDeclarePassive(stagingQueueOrTargetQueue);
+        LOG.debug("Declaring worker queue {}...", stagingQueueOrTargetQueue);
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java:121]: How about a static code block to initialize queueArguments?

**Review by andyreidz — CHANGES_REQUESTED** [review]

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java [hunk 5]
```diff
@@ -60,11 +60,22 @@ public final class QueueServicesFactory
...
-    public static final String RABBIT_PROP_KEY_MAX_PRIORITY = "x-max-priority";
+    private static final String RABBIT_PROP_KEY_MAX_PRIORITY = "x-max-priority";
 
-    public static final String RABBIT_PROP_QUEUE_TYPE = "x-queue-type";
+    private static final String RABBIT_PROP_QUEUE_TYPE = "x-queue-type";
 
-    public static final String RABBIT_PROP_QUEUE_TYPE_CLASSIC = "classic";
+    private static final String RABBIT_PROP_QUEUE_TYPE_CLASSIC = "classic";
+
+    private static final Map<String, Object> queueArguments = new HashMap<>();
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]:
```suggestion
    private static final Map<String, Object> QUEUE_ARGUMENTS = new HashMap<>();
```

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java [hunk 6]
```diff
@@ -59,6 +60,23 @@ public final class QueueServicesFactory
...
+
+    private static final Map<String, Object> QUEUE_ARGUMENTS = new HashMap<>();
+
+    static {
+        final int queueMaxPriority = ScheduledExecutorConfig.getQueueMaxPriority();
+        final String queueType = ScheduledExecutorConfig.getQueueType();
+        if (queueMaxPriority > 0 && queueType.equals(RABBIT_PROP_QUEUE_TYPE_CLASSIC)) {
+            queueArguments.put(RABBIT_PROP_KEY_MAX_PRIORITY, queueMaxPriority);
+        }
+        queueArguments.put(RABBIT_PROP_QUEUE_TYPE, queueType);
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]: 😭

Comment (raghavgopal) [reply job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServicesFactory.java]: I noticed and was just making this change before you commented .

**Review by andyreidz — APPROVED** [review]

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/ScheduledExecutorConfig.java [hunk 7]
```diff
@@ -118,6 +118,22 @@ public static int getDropTablesSchedulerPeriod() {
         return Integer.parseInt(period);
     }
 
+    public static String getQueueType() {
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/ScheduledExecutorConfig.java:121]: Update the readme with these new env's and their defaults.
https://github.com/JobService/job-service/blob/develop/job-service-scheduled-executor/README.md

**Pull request merged**